pandas/core/algorithms.py

```diff
@@ -310,7 +310,7 @@ def _check_object_for_strings(values: np.ndarray) -> str:
 # --------------- #
 
 
-def unique(values):
+def unique(values, mask=None):
```

Comment:
Not sure if this is a good idea, since this changes the api. A better alternative would probably be a _unique function that can take a mask

Reply:
agreed

also can annotate mask?

Reply:
Yep, will do when moving to a private function

Reply:
Done